src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -106,7 +106,7 @@ def create_keyvault_data_plane_client(cli_ctx):
     version = str(get_api_version(cli_ctx, ResourceType.DATA_KEYVAULT))
 
     def get_token(server, resource, scope):  # pylint: disable=unused-argument
-        return Profile(cli_ctx=cli_ctx).get_login_credentials(resource)[0]._token_retriever()  # pylint: disable=protected-access
+        return Profile(cli_ctx=cli_ctx).get_raw_token(resource)[0]
```

Comment:
Can we make `Profile` object a global singleton?

Reply:
Good suggestion. Will consider doing that in the future.